lib/src/launchpad.rs

```diff
@@ -238,6 +248,255 @@ impl LaunchpadMk2 {
     }
 }
 
+impl LaunchpadPro {
+    /// Attempt to find the first Launchpad Pro by scanning
+    /// available MIDI ports with matching names
+    pub fn guess() -> LaunchpadPro {
+        let midi = pm::PortMidi::new().expect("Failed to open PortMidi Instance!");
+        let mut retval = Self::guess_from(&midi);
+        retval.midi = Some(midi);
+        retval
+    }
+
+    /// Attempt to find the first Launchpad Pro by scanning
+    /// available MIDI ports with matching names. Bring your own
+    /// PortMidi.
+    pub fn guess_from(midi: &pm::PortMidi) -> LaunchpadPro {
+        let devs = midi.devices().expect("Failed to get Midi Device!");
+
+        let mut input: Option<i32> = None;
+        let mut output: Option<i32> = None;
+
+        for d in devs {
+            if !d.name().contains("Launchpad Pro Standalone") {
+                continue;
+            }
+
+            if input.is_none() && d.is_input() {
+                input = Some(d.id() as i32);
+            }
+
+            if output.is_none() && d.is_output() {
+                output = Some(d.id() as i32);
+            }
+
+            if input.is_some() && output.is_some() {
+                break;
+            }
+        }
+
+        let input_port = input.expect("No Launchpad Pro Input Found!");
+        let output_port = output.expect("No Launchpad Pro Output Found!");
+
+        let input_device = midi.device(input_port)
+            .expect("No Launchpad Pro Input found!");
+        let output_device = midi.device(output_port)
+            .expect("No Launchpad Pro Output found!");
+
+        let input = midi.input_port(input_device, 1024)
+            .expect("Failed to open port");
+        let output = midi.output_port(output_device, 1024)
+            .expect("Failed to open port");
+
+        LaunchpadPro {
+            input_port: input,
+            output_port: output,
+            midi: None,
+        }
+    }
+
+    /// Set all LEDs to the same color
+    pub fn light_all(&mut self, color: Color) {
+        assert_color(color);
+        // F0h 00h 20h 29h 02h 18h 0Eh <Colour> F7h
+        // Message cannot be repeated.
+        self.output_port
+            .write_sysex(0, &[0xF0, 0x00, 0x20, 0x29, 0x02, 0x10, 0x0E, color, 0xF7])
+            .expect("Fail");
+    }
+
+    /// Set a single LED to flash. Uses a smaller header than `flash_led` or
+    /// `flash_leds` with a single item
+    pub fn flash_single(&mut self, led: &ColorLed) {
```

Comment:
The message for flashing LEDs can take up to 97 LEDs, so this should probably be updated to have a similar pattern to `light_leds`/`light_led` to be able to send out the flash LED message with chunks of up to 97 LEDs.